Rakefile

```
@@ -262,6 +263,397 @@ task :xcode => [:dependencies] do
+      task :check_oss do
+        tool_check(oss_tools)
+      end
+
+      task :check_developer do
+        tool_check(developer_tools)
+      end
+
+      #check if the developer tool is present in the machine, if not install
+      def tool_check(hash)
+        hash.each do |key, value|
+          puts "Checking system for #{key}"
+          unless command?(key)
+            tool_install(value)
+          else
+            puts "#{key} found"
+          end
+        end
+      end
+
+      #install selected developer tool
+      def tool_install(tool)
+        puts "#{tool} not found.  Installing #{tool}"
+        sh "brew install #{tool}"
+      end
+    end
+  end
+
+  namespace :lint do
+    task :check do
+      if !git_initialized?
+        puts "Initializing git repository"
+        sh "git init", verbose: false
+      end
+
+      Rake::Task["git:install_hooks"].invoke
+    end
+
+    def git_initialized?
+      sh "git rev-parse --is-inside-work-tree > /dev/null 2>&1", verbose: false
+    end
+  end
+end
+
+#Credentials deals with the setting up the developer's WPCOM API app ID and app Secret
+namespace :credentials do
+  task :setup => %w[credentials:prompt credentials:set_app_secrets]
+
+  task :prompt do
+    puts ""
+    puts "====================================================================================="
+    puts "To be able to log into the WordPress app while developing you will need to setup API credentials"
+    puts "To do this follow these steps"
+    puts ""
+    puts ""
+    puts ""
+    puts "====================================================================================="
+
+    puts "1. Go to https://wordpress.com/start/user and create a WordPress.com account (if you don't already have one)."
+    prompt_for_continue("Once you have created your account,")
+
+    puts "====================================================================================="
+    puts "2. Now register an API application at https://developer.wordpress.com/apps/."
+    prompt_for_continue("Once you have registered your API App,")
+
+    puts "====================================================================================="
+    puts '3. Make sure to set "Redirect URLs"= https://localhost and "Type" = Native and click "Create" then "Update".'
+    prompt_for_continue("Once you have set the redirect url and type,")
+
+    puts "====================================================================================="
+    prompt_for_continue("Lastly, Keep your Client ID and App Secret on hand for the next steps,")
+  end
+
+  def prompt_for_continue(prompt)
+    puts "#{prompt} Please press enter to continue"
+    STDIN.gets.strip
+  end
+
+  #user given app id and secret and create a new wpcom_app_credentials file
+  task :set_app_secrets do
+    create_secrets_file
+    set_app_secrets(get_client_id, get_client_secret)
+    remove_temp_file
+  end
+
+  def get_client_id
+    STDOUT.puts "Please enter your Client id"
```

Just a nitpick. You don't have to address this here, it an easily be a followup PR because it doesn't have to do with the actual workings of the code.

There's a bit of inconsistencies in how the "Client ID" and "Client Secret" are typed. Sometimes they are "Client ID", other times "client ID", or "Client id", or "client id". It would be great to have it consistent. To be honest, though, [our docs](https://developer.wordpress.com/docs/oauth2/) are not consistent with it either... 😳 

```suggestion
    STDOUT.puts "Please enter your Client ID"
```